FlowersStore/Scenes/FlowersList/FlowersListView.swift

```diff
@@ -7,22 +7,85 @@
 
 import UIKit
 
+protocol FlowersListViewDelegate: AnyObject {
+    func flowersListView(_ flowersListView: FlowersListView, didTapFavorite flower: Flower)
+}
+
 final class FlowersListView: UIView {
     // MARK: - Properties
+    private let flowers: [Flower]
+    private(set) var indexPathForFavoriteItem: IndexPath?
+    private weak var delegate: FlowersListViewDelegate?
     
     // MARK: - UI Components
-    private lazy var collectionView: UICollectionView = {
-        let collectionView: UICollectionView = .init()
-        
+    private(set) lazy var collectionView: UICollectionView = {
+        let layout: FlowersListCollectionViewFlowLayout = .init()
+        let collectionView: UICollectionView = .init(frame: .zero, collectionViewLayout: layout)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.delegate = self
+        collectionView.dataSource = self
         return collectionView
     }()
     
     // MARK: - Initializer Methods
-    override init(frame: CGRect) {
-        super.init(frame: frame)
+    init(
+        flowers: [Flower],
+        delegate: FlowersListViewDelegate?
+    ) {
+        self.flowers = flowers
+        self.delegate = delegate
+        super.init(frame: .zero)
+        setupView()
     }
     
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
+    
+    // MARK: - Public Methods
+    func updateFavorite() {
```

Comment:
<!--
  0 failure: 
  1 warning:  Function 'updateF...
  
  
  DangerID: danger-id-Danger;
  File: FlowersStore/Scenes/FlowersList/FlowersListView.swift;
  Line: 46;
-->

- :warning: Function 'updateFavorite()' is unused